### PR TITLE
fix: add code to access modes when creating a group with guests AR-3188

### PR DESCRIPTION
**Pull request by tmspzz**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3188" title="AR-3188" target="_blank">BugAR-3188</a>  When we create a group on AR, we can not enable guestlinks
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When creating a group, guest links can't be created

### Solutions

When creating a group where guest are allowed add access mode `code` and `invite` to the group 

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Create a group from the app with guests enabled, proceed to generate a guest link and notice it works.

### Notes (Optional)

Additional changes are needed when toggling guests on and off after creation. This will follow in a separate PR.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt [hunk 1]
```diff
@@ -193,7 +193,10 @@ class NewConversationViewModel @Inject constructor(
                 options = ConversationOptions().copy(
                     protocol = newGroupState.groupProtocol,
                     readReceiptsEnabled = groupOptionsState.isReadReceiptEnabled,
-                    accessRole = groupOptionsState.accessRoleState
+                    accessRole = groupOptionsState.accessRoleState,
+                    access = if (groupOptionsState.accessRoleState.contains(Conversation.AccessRole.GUEST)) {
+                        setOf(Conversation.Access.INVITE, Conversation.Access.CODE)
```
Comment (tmspzz) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:198]: This is the actual change

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1536/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5930365)(5930365)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5930365)(5930365)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:198]: Does that mean that if we later update this `AccessRole` using `UpdateConversationAccessRoleUseCase` then it should also add or remove `Access` depending on the newly selected `AccessRole` or should `Access` be set only when creating the group?

Comment (tmspzz) [reply app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:198]: Yes, that is the next PR

File: app/src/test/kotlin/com/wire/android/ui/authentication/login/email/LoginEmailViewModelTest.kt [hunk 2]
```diff
@@ -18,6 +18,8 @@
  *
  */
 
+@file:Suppress("MaxLineLength")
```
Comment (gongracr) [inline app/src/test/kotlin/com/wire/android/ui/authentication/login/email/LoginEmailViewModelTest.kt:21]: I don't see why with these changes this suppress will be needed as they all seem to respect the max line length 🤷

File: app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelArrangement.kt [hunk 3]
```diff
@@ -225,5 +224,27 @@ internal class NewConversationViewModelArrangement {
         coEvery { isSelfTeamMember() } returns result
     }
 
+    fun withGuestEnabled(x: Boolean) = apply {
```
Comment (gongracr) [inline app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelArrangement.kt]: just a tiny readability suggestion
```suggestion
    fun withGuestEnabled(isGuestModeEnabled: Boolean) = apply {
```

File: app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelArrangement.kt [hunk 4]
```diff
@@ -225,5 +224,27 @@ internal class NewConversationViewModelArrangement {
...
+                add(Conversation.AccessRole.NON_TEAM_MEMBER)
+            } else {
+                remove(Conversation.AccessRole.GUEST)
+                remove(Conversation.AccessRole.NON_TEAM_MEMBER)
+            }
+        }
+        viewModel.groupOptionsState = viewModel.groupOptionsState.copy(isAllowGuestEnabled = x, accessRoleState = newAccessRole)
+    }
+
+    fun withServicesEnabled(x: Boolean) = apply {
```
Comment (gongracr) [inline app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelArrangement.kt]: Same as above
```suggestion
    fun withServicesEnabled(areServicesEnabled: Boolean) = apply {
```

File: app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelTest.kt [hunk 5]
```diff
@@ -169,6 +172,34 @@ class NewConversationViewModelTest {
...
+                    setOf(Conversation.Access.INVITE, Conversation.Access.CODE),
+                    setOf(Conversation.AccessRole.TEAM_MEMBER, Conversation.AccessRole.NON_TEAM_MEMBER, Conversation.AccessRole.GUEST),
+                    true,
+                    ConversationOptions.Protocol.PROTEUS,
+                    null
+                )
+            )
+        }
+    }
+
```
Comment (gongracr) [inline app/src/test/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModelTest.kt:202]: Should we add another test checking that if `isGuestEnabled = false` then `ConversationOptions.access` will be empty?

**Review by gongracr — APPROVED** [review]
Looking good 💻 🍎  I added some tiny non-blocking comments for the extra coins ⭐

**Review by saleniuk — APPROVED** [review]

File: app/src/test/kotlin/com/wire/android/ui/authentication/login/email/LoginEmailViewModelTest.kt (hunk 2 above)
Comment (tmspzz) [reply app/src/test/kotlin/com/wire/android/ui/authentication/login/email/LoginEmailViewModelTest.kt:21]: They don't tho :D

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1536/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1536/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5931601)(5931601)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5931601)(5931601)-staging-compat.apk)

**Pull request merged**